src/cpp/benders/logger/MathLogger.cpp

```diff
@@ -0,0 +1,98 @@
+#include "logger/MathLogger.h"
+
+#include <iomanip>
+#include <sstream>
+
+void MathLoggerBase::setHeadersList() {
+  MathLogger::setHeadersList({ITE, LB, UB, BESTUB, ABSOLUTE_GAP, RELATIVE_GAP,
+                              MINSIMPLEX, MAXSIMPLEX, TIMEMASTER,
+                              SUB_PROBLEMS_TIME_CPU, SUB_PROBLEMS_TIME_WALL});
+}
+
+void MathLogger::setHeadersList(const std::list<std::string>& headers) {
+  headers_.clear();
+  headers_ = headers;
+}
+
+void MathLogger::write_header() {
+  setHeadersList();
+  for (const auto& header : Headers()) {
+    TheLogDestination() << header;
+  }
+  TheLogDestination() << std::endl;
+}
+void MathLoggerBase::Print(const CurrentIterationData& data) {
+  TheLogDestination() << Indent(10) << data.it;
+  if (data.lb == -1e20)
+    TheLogDestination() << Indent(20) << "-INF";
```

Comment:
I prefer that we really print `1e+20` and `1e-20` in the math logs as these are the actual values of the data. the purpose of the math logs is to monitor exactly what happens in the data and not to have an interpretation of it as infinity

Reply:
you say, i do